enterprise/cli/provisionerdaemonstart.go

```diff
@@ -104,6 +104,22 @@ func (r *RootCmd) provisionerDaemonStart() *serpent.Command {
 				return err
 			}
 
+			displayedTags := make(map[string]string)
+			for key, val := range tags {
+				displayedTags[key] = val
+			}
+
+			if provisionerKey != "" {
+				pkDetails, err := client.GetProvisionerKey(ctx, provisionerKey)
+				if err != nil {
+					return xerrors.New("unable to get provisioner key details")
+				}
+
+				for k, v := range pkDetails.Tags {
+					displayedTags[k] = v
+				}
```

Comment:
I think we can overwrite `displayedTags` here entirely. If using a provisioner key, none of the tags specified via `tags` can be used. (In fact, I don't think you're even _allowed_ specify both `--key` and `--tag`)

Reply:
I preferred to keep two different variables as otherwise it would require some other logic changes.

The tag variable is used to call the `/provisionerdaemon/serve` endpoint below, and using the same variable would mean changes here too, which I feel like we dont want. 

wdyt ?

Reply:
For the sake of correctness, I'd like to make this change.
Otherwise this may confuse future readers of the code.
However, let's do it as a separate follow-up PR.